examples/cpp/demo_config.cpp

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (2019) Cobalt Speech and Language, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "demo_config.h"
+
+#include <algorithm>
+#include <cctype>
+#include <cpptoml.h>
+#include <stdexcept>
+
+
+// Converts the Cubic encoding from a string to the enum value.
+cobaltspeech::cubic::RecognitionConfig::Encoding
+   convertEncoding(std::string encodeStr)
+{
+    // First make the encoding string lowercase
+    std::transform(encodeStr.begin(), encodeStr.end(), encodeStr.begin(),
+        [](unsigned char c){ return std::tolower(c); });
+
+    if (encodeStr == "raw_linear16")
+    {
+        return cobaltspeech::cubic::RecognitionConfig::RAW_LINEAR16;
+    }
+    if (encodeStr == "wav")
+    {
+        return cobaltspeech::cubic::RecognitionConfig::WAV;
+    }
+    if (encodeStr == "mp3")
+    {
+        return cobaltspeech::cubic::RecognitionConfig::MP3;
+    }
+    if (encodeStr == "flac")
+    {
+        return cobaltspeech::cubic::RecognitionConfig::FLAC;
+    }
+    if (encodeStr == "vox8000")
+    {
+        return cobaltspeech::cubic::RecognitionConfig::VOX8000;
+    }
+    if (encodeStr == "ulaw8000")
+    {
+        return cobaltspeech::cubic::RecognitionConfig::ULAW8000;
+    }
+
+    // If we made it this far, we have an unsupported encoding
+    throw std::runtime_error("unsupported encoding " + encodeStr);
+}
+
+DemoConfig DemoConfig::parseApplicationArgs(int argc, char *argv[])
+{
+    std::string configFilename;
+
+    if (argc == 1)
+    {
+        // No arguments were specified. Use a default config filename.
+        configFilename = "config.toml";
+    }
```

Comment:
Do we want to have this?

Reply:
I could go either way. When I use it, I tend to always specify the path to my config file, so I don't have a problem making it a required arg.